src/bin/nydusd/main.rs

```diff
@@ -363,6 +373,83 @@ fn handle_rlimit_nofile_option(args: &ArgMatches, option_name: &str) -> Result<(
     Ok(())
 }
 
+fn send_unix_socket_request(socket_path: &str, method: &str, url: &str) -> Option<String> {
+    let mut stream = match UnixStream::connect(socket_path).ok() {
+        Some(stream) => stream,
+        None => return None,
```

Comment:
We'd better return a `Result` instead of an `Option`, and handle the `Result` outside.